lib/hammer_cli/abstract.rb

```diff
@@ -341,27 +341,27 @@ def exception_handler_class
     end
 
     def self.desc(desc=nil)
-      @desc = desc if desc
-      @desc
+      return @desc ||= nil unless desc
```

Comment:
I think this is a lot harder to read. It's also a bit of an odd API where it's both a setter and a getter.

Can you separate the undefined variable changes to a separate commit? AFAIK they're warnings, not errors so it should be possible to update rake without addressing them immediately.